File: solutions/uri/2059/2059.cpp
#include <cstdint>
#include <iostream>

int main() {
    int16_t p, j1, j2, r, a, winner;

    while (std::cin >> p >> j1 >> j2 >> r >> a) {
        if ((p == 1 && (j1 + j2) % 2 == 0) || (p == 0 && (j1 + j2) % 2 == 1)) {
            winner = 1;
        } else {
            winner = 2;
        }

        if (r == 1 && a == 1) {
            winner = 2;
        } else if (r == 0 && a == 1) {
            winner = 1;
        } else if (r == 1) {
            winner = 1;
        }

        std::cout << "Jogador " << winner << " ganha!" << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/2059/2059.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while line := input().strip():
    p, j1, j2, r, a = map(int, line.split())

    if (p == 1 and (j1 + j2) % 2 == 0) or (p == 0 and (j1 + j2) % 2 == 1):
        winner = 1
    else:
        winner = 2

    if r == 1 and a == 1:
        winner = 2
    elif r == 0 and a == 1:
        winner = 1
    elif r == 1:
        winner = 1

    output(f'Jogador {winner} ganha!\n')
